pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -786,6 +730,73 @@ func CheckForExpiredNodeLeases(nodeList *corev1.NodeList, leaseList *coordinatio
 	return nil
 }
 
+func CheckNodesScaling(ctx context.Context, seedClient client.Client, nodeList *corev1.NodeList, machineDeploymentList *machinev1alpha1.MachineDeploymentList, controlPlaneNamespace string) (string, error) {
+	var (
+		readyNodes          int
+		registeredNodes     = len(nodeList.Items)
+		desiredMachines     = getDesiredMachineCount(machineDeploymentList.Items)
+		nodeNotManagedByMCM int
+	)
+
+	for _, node := range nodeList.Items {
+		if metav1.HasAnnotation(node.ObjectMeta, annotationKeyNotManagedByMCM) && node.Annotations[annotationKeyNotManagedByMCM] == "1" {
+			nodeNotManagedByMCM++
+			continue
+		}
+		if node.Spec.Unschedulable {
+			continue
+		}
+		for _, condition := range node.Status.Conditions {
+			if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionTrue {
+				readyNodes++
+			}
+		}
+	}
+
+	// only nodes that are managed by MCM is considered
+	registeredNodes = registeredNodes - nodeNotManagedByMCM
+
+	machineList := &machinev1alpha1.MachineList{}
+	if registeredNodes == desiredMachines && readyNodes == desiredMachines {
```

Comment:
isn't  `registeredNodes` >= `readyNodes` so:
`registeredNodes == desiredMachines && readyNodes == desiredMachines`  if and only if `readyNodes == desiredMachines` ?